esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -3595,7 +3595,14 @@ bool CWsDfuEx::addToLogicalFileList(IPropertyTree& file, const char* nodeGroup,
 
         int numSubFiles = file.hasProp(getDFUQResultFieldName(DFUQRFnumsubfiles));
         if(numSubFiles)
+        {
             lFile->setIsSuperfile(true);
+            if (version >= 1.52)
+            {
+                numSubFiles = file.getPropInt(getDFUQResultFieldName(DFUQRFnumsubfiles));
+                lFile->setNumOfSubfiles(numSubFiles);
```

Comment:
Is this directly related, strictly necessary for this fix?
Or is it just new meta info in the reply to be used by a client/future purpose?

Reply:
The UI needs to know the number of subfiles in order to enable/disable the  'Retain Superfile Structure' checkbox.

Reply:
right makes sense, but I don't think I see where eclwatch is using NumOfSubfiles ?

Reply:
ECLWatch has not implemented the function yet: https://track.hpccsystems.com/browse/HPCC-22186

Reply:
ok, I've linked the issues.